lnwire/dns_addr.go

```diff
@@ -0,0 +1,146 @@
+package lnwire
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+// DNSAddr is a custom implementation of the net.Addr interface.
+type DNSAddr struct {
+	// Hostname represents the DNS hostname (e.g., "example.com").
+	Hostname string
+
+	// Port is the network port number (e.g., 9735 for LND peer port).
+	Port int
+}
+
+// A compile-time check to ensure that DNSAddr implements the net.Addr
+// interface.
+var _ net.Addr = (*DNSAddr)(nil)
+
+// Network returns the network type, e.g., "tcp".
+func (d *DNSAddr) Network() string {
+	return "tcp"
+}
+
+// String returns the address in the form "hostname:port".
+func (d *DNSAddr) String() string {
+	return net.JoinHostPort(d.Hostname, strconv.Itoa(d.Port))
+}
+
+// NewDNSAddr creates a new DNS address with the given host and port. It returns
+// an error if the hostname is invalid according to DNS standards/BOLT 07
+// specifications or if the port is outside the valid TCP
+// range (1-65535).
+func NewDNSAddr(host string, port int) (*DNSAddr, error) {
+	dnsAddr := &DNSAddr{
+		Hostname: host,
+		Port:     port,
+	}
+
+	if err := dnsAddr.validate(); err != nil {
+		return nil, err
+	}
+
+	return dnsAddr, nil
+}
+
+// validate checks that the DNSHostnameAddress is well-formed according to
+// Lightning Network's BOLT 07 specification requirements. It ensures the
+// hostname is not empty, doesn't exceed 255 characters, contains only ASCII
+// characters, and that the port number is within the valid TCP range (1-65535).
+func (d *DNSAddr) validate() error {
```

Comment:
i still dont think we need all this validation on this layer. I think it's ok to do all the validation for our _own_ users hostname string at config load time or on the update-nod-ann rpc functionality. But the spec only says "The hostname must be ASCII and not longer than 258 chars" so imo that's the only validation to do here. 
wdyt @yyforyongyu ?